net/src/stream/mod.rs

```diff
@@ -21,17 +20,15 @@ use pin_project_lite::pin_project;
 
 use karyon_core::async_runtime::io::{AsyncRead, AsyncWrite};
 
-use crate::codec::{Decoder, Encoder};
+use crate::codec::{Buffer, ByteBuffer, Decoder, Encoder};
 
-use buffer::Buffer;
-
-const BUFFER_SIZE: usize = 2048 * 2048; // 4MB
+const BUFFER_SIZE: usize = 4096 * 4096; // 16MB
```

Comment:
The problem is that that actual allocated memory will be much higher if the value increase, and it might not be needed for most apps. I would rather keep it low and have a compile time or environment override for the applications that need it. It also fixes the issue of "16MB is not enough".